lib/devicetrust/native/device_windows.go

```diff
@@ -0,0 +1,489 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package native
+
+import (
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"math/big"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"strings"
+
+	"github.com/google/go-attestation/attest"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	"github.com/gravitational/teleport/lib/devicetrust"
+)
+
+const (
+	deviceStateFolderName  = ".teleport-device"
+	attestationKeyFileName = "attestation.key"
+)
+
+// Ensures that device state directory exists.
+// It returns the absolute path to where the attestation key can be found:
+// ~/teleport-device/attestation.key
+func setupDeviceStateDir(getHomeDir func() (string, error)) (string, error) {
+	home, err := getHomeDir()
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	deviceStateDirPath := path.Join(home, deviceStateFolderName)
+	keyPath := path.Join(deviceStateDirPath, attestationKeyFileName)
+
+	_, err = os.Stat(deviceStateDirPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			// If it doesn't exist, we can create it and return as we know
+			// the perms are correct as we created it.
+			if err := os.Mkdir(deviceStateDirPath, 700); err != nil {
+				return "", trace.Wrap(err)
+			}
+			return keyPath, nil
+		}
+		return "", trace.Wrap(err)
+	}
+
+	return keyPath, nil
+}
+
+func openTPM() (*attest.TPM, error) {
+	cfg := &attest.OpenConfig{
+		TPMVersion: attest.TPMVersion20,
+	}
+
+	tpm, err := attest.OpenTPM(cfg)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return tpm, nil
+}
+
+// getMarshaledEK returns the EK public key in PKIX, ASN.1 DER format.
+func getMarshaledEK(tpm *attest.TPM) ([]byte, error) {
+	eks, err := tpm.EKs()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if len(eks) == 0 {
+		return nil, trace.BadParameter("no endorsement keys found in tpm")
+	}
+	// TODO(noah): Marshal EK Certificate instead of key if present:
+	// https://github.com/gravitational/teleport.e/issues/1393
+	encodedEK, err := x509.MarshalPKIXPublicKey(eks[0].Public)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return encodedEK, nil
+}
+
+// loadOrCreateAK attempts to load an AK from disk. A NotFound error will be
```

Comment:
```suggestion
// loadAK attempts to load an AK from disk. A NotFound error will be
```

assuming it just loads and never creates?